fontawesome_markdown/main.py

```diff
@@ -3,22 +3,46 @@
 from markdown.inlinepatterns import Pattern
 from markdown.util import etree
 from .icon_list import icons
+from .styleutil import prefix_to_style, style_to_prefix
+import json
 
-fontawesome_pattern = r':fa-([-\w]+):'
+fontawesome_pattern = r':(fa[bsrl]?)?\s?fa-([-\w]+)\s?(fa-(xs|sm|lg|[\d+]x|10x))?:'
 
 
 class FontAwesomeException(Exception):
-    'Exception for unknown icon names'
+    'Exception for unknown icon name, prefix or size'
     pass
 
 
 class FontAwesomePattern(Pattern):
     'Markdown pattern class for matching things that look like FA icons'
     def handleMatch(self, m):
         el = etree.Element('i')
-        icon_name = m.group(2)
+        prefix = m.group(2)
+        icon_name = m.group(3)
+        size = m.group(4)
         if icon_name in icons:
-            el.attrib = {'class': 'fa fa-{0}'.format(icon_name)}
+            styles = icons[icon_name]
+            if not prefix and 'solid' in styles:
+                # default prefix is fa
+                prefix = 'fa'
+            elif not prefix and 'solid' not in styles:
+                style = styles[0]
+                prefix = style_to_prefix(style)
+                if not prefix:
+                    message = "unknown style {0}".format(style)
+                    raise FontAwesomeException(message)
+            
+            elif prefix and prefix_to_style(prefix) not in styles:
+                message = "{0} have not prefix '{1}'.\n"
+                "Allowed prefix is {2} ".format(icon_name, prefix, styles)
+                
+                raise FontAwesomeException(message)
+            
+            clazz = '{0} fa-{1}'.format(prefix, icon_name)
```

Comment:
Just call it `css_class`

Reply:
`klass` and `clazz` are conventionally used for variables that hold Python classes